src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -68,6 +68,16 @@ def validate_keyvault(cmd, namespace):
                                           'vaults', 'Microsoft.KeyVault')
 
 
+def validate_vm_name_for_monitor_metrics(cmd, namespace):
+    if hasattr(namespace, 'resource'):
+        namespace.resource = _get_resource_id(cmd.cli_ctx, namespace.resource, namespace.resource_group_name,
+                                              'virtualMachines', 'Microsoft.Compute')
+    elif hasattr(namespace, 'resource_uri'):
+        namespace.resource_uri = _get_resource_id(cmd.cli_ctx, namespace.resource_uri, namespace.resource_group_name,
+                                                  'virtualMachines', 'Microsoft.Compute')
+    del namespace.resource_group_name
```

Comment:
Curious about why you delete rg name

Reply:
If we don't delete it, we will meet errors